connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -143,7 +144,7 @@ public synchronized Future<Void> executeFailed(Stage stage, Class<?> executingCl
      * @param <V> return type of the result of the operation.
      * @return result of the operation
      */
-    public synchronized <V> V execute(Operation<V> operation, Stage stage, Class<?> executingClass) {
+    public <V> V execute(ProcessingContext<?> context, Operation<V> operation, Stage stage, Class<?> executingClass) {
```

Comment:
I know this isn't your fault but if you have time, could we add a `throws` clause to the Javadocs stating that an exception will be thrown if a non-retriable error is encountered? I always get tripped up reading interactions with this class and a big part of it is trying to understand the conditions where exceptions are thrown, the context is marked as failed, or `null` is returned.